python/openassetio/test/manager/_implementation.py

```diff
@@ -0,0 +1,151 @@
+#
+#   Copyright 2013-2021 The Foundry Visionmongers Ltd
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+"""
+Boilerplate classes for initialization and execution of test cases.
+"""
+import unittest
+
+from openassetio import hostAPI, pluginSystem, logging
+
+
+__all__ = ['createHarness']
+
+
+def createHarness(managerIdentifier):
+    """
+    Create the test harness used begin test case execution.
+    """
+    hostInterface = _ValidatorHarnessHostInterface()
+    logger = logging.SeverityFilter(logging.ConsoleLogger())
+    managerFactory = pluginSystem.PluginSystemManagerFactory(logger)
+
+    session = hostAPI.Session(hostInterface, logger, managerFactory)
+    session.useManager(managerIdentifier)
+
+    loader = _ValidatorTestLoader(session)
+    return _ValidatorHarness(unittest.main, loader)
+
+
+class _ValidatorHarness:
+    """
+    The manager test harness.
+
+    It acts as an anti-corruption layer around a unittest-compatible
+    test runner, injecting a custom testcase loader.
+    """
+    # pylint: disable=too-few-public-methods
+
+    _kValidatorHarnessProgramName = "manager"
```

Comment:
[consider] Similar to other comment, this is for argparse help messages, so that e.g.
```
$ OPENASSETIO_PLUGIN_PATH=resources/examples/manager/SampleAssetManager/python python -m openassetio.test.manager -f resources/examples/manager/SampleAssetManager/test/fixtures.py --some-unsupported-switch
usage: manager [-h] [-v] [-q] [--locals] [-f] [-c] [-b] [-k TESTNAMEPATTERNS] [tests [tests ...]]
manager: error: unrecognized arguments: --some-unsupported-switch
```
so we probably want something a bit more descriptive than `manager`.

Reply:
Fixed [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/184/commits/6699dbda1241566581b328ceda68e93ec22d2134).